libgalois/src/PropertyGraph.cpp

```diff
@@ -586,33 +586,38 @@ katana::Result<void>
 katana::PropertyGraph::DoWrite(
     tsuba::RDGHandle handle, const std::string& command_line,
     tsuba::RDG::RDGVersioningPolicy versioning_action) {
-  std::unique_ptr<tsuba::FileFrame> topology_res = nullptr;
-  std::unique_ptr<tsuba::FileFrame> node_entity_type_id_array_res = nullptr;
-  std::unique_ptr<tsuba::FileFrame> edge_entity_type_id_array_res = nullptr;
-
-  if (!rdg_.topology_file_storage().Valid()) {
-    auto res = WriteTopology(topology());
-    if (!res) {
-      return res.error();
-    }
-    topology_res = std::move(res.value());
+  if (rdg_.topology_file_storage().Valid() ||
+      rdg_.node_entity_type_id_array_file_storage().Valid() ||
+      rdg_.edge_entity_type_id_array_file_storage().Valid()) {
+    return KATANA_ERROR(
+        ErrorCode::AssertionFailed,
+        "All file storage files must be valid. Topology file valid: {}, Node "
+        "Entity Type ID Array valid: {}, Edge Entity Type ID Array valid: {}",
+        rdg_.topology_file_storage().Valid(),
+        rdg_.node_entity_type_id_array_file_storage().Valid(),
+        rdg_.edge_entity_type_id_array_file_storage().Valid());
   }
 
-  if (!rdg_.node_entity_type_id_array_file_storage().Valid()) {
-    auto res = WriteEntityTypeIDsArray(node_entity_type_ids_);
-    if (!res) {
-      return res.error();
-    }
-    node_entity_type_id_array_res = std::move(res.value());
+  auto topology_write = WriteTopology(topology());
+  if (!topology_write) {
+    return topology_write.error();
```

Comment:
While we're here this whole block could just be:
```
std::unique_ptr<tsuba::FileFrame> topology_res = KATANA_CHECKED(WriteTopology(topology()));
```

Ditto with those below; TBC what you have here is fine, just more verbose that necessary 😄

Reply:
yes, in fact, I had mentioned the same thing in a comment on a preview PR.